Source/Core/Core/State.cpp

```diff
@@ -280,8 +281,13 @@ struct CompressAndDumpState_args
 static void CompressAndDumpState(CompressAndDumpState_args save_args)
 {
 	std::lock_guard<std::mutex> lk(*save_args.buffer_mutex);
-	if (!save_args.wait)
+
+	Common::ScopeGuard on_exit([]()
```

Comment:
I'd like to have a comment saying something like that g_compressAndDumpStateSyncEvent must not be set before the file object's destructor runs, so that it's clear why the ScopeGuard is being used instead of the old solution.